src/client/impls/reqwest.rs

```
@@ -0,0 +1,534 @@
+    async fn search_article_page(
+        &self,
+        name: &ArticleTitle,
+        page: NonZeroUsize,
+    ) -> Res<Vec<RecommendedArticle>> {
+        const ENDPOINT: &str = "/api/articles/search";
+        let mut url = self.base_url().join(ENDPOINT).expect(ANGRY_URL);
+        url.query_pairs_mut()
+            .append_pair("name", name.as_ref())
+            .append_pair("page", &page.to_string());
+        let response = send_ok!(self.get(url));
+        let articles = json_ok!(response, Vec<RecommendedArticle>);
+        Ok(articles)
+    }
+
+    #[tracing::instrument(name = "Loading followers")]
+    async fn get_followers_page(&self, id: &UserId, page: NonZeroUsize) -> Res<Vec<FollowerUser>> {
+        const ENDPOINT: &str = "/api/relationships/";
+        let mut url = self
+            .base_url()
+            .join(ENDPOINT)
+            .and_then(|endpoint| endpoint.join(&format!("{}/followers", id)))
+            .expect(ANGRY_URL);
+        url.query_pairs_mut().append_pair("page", &page.to_string());
+        let response = send_ok!(self.get(url));
+        let followers = json_ok!(response, Vec<FollowerUser>);
+        Ok(followers)
+    }
+
+    #[tracing::instrument(name = "Getting replies")]
+    async fn get_replies(&self, comment: &CommentId) -> Res<Vec<ReplyComment>> {
+        const ENDPOINT: &str = "/api/articles/000000000000000000000000/comments/";
+        let url = self
+            .base_url()
+            .join(ENDPOINT)
+            .and_then(|article_comments| article_comments.join(&format!("{}/replies", comment)))
+            .expect(ANGRY_URL);
+        let response = send_ok!(self.get(url));
+
+        if response.status() == StatusCode::UNAUTHORIZED {
+            return Err(Error::NoObject);
+        }
+        // TODO add assertions for expected response code in all of the functions
+        assert_eq!(response.status(), StatusCode::OK, "Unexpected status code");
```

Are u sure about needing to paste `assert_eq` here?

---

Maybe I'm missing smth, but it will panic

---

yeah, It will, I thought I would be a good idea to panic in case of unexpected response

But now, when I thing about it, I should probably add one more variant to error, representing this situation. Thanks for bringing it up.